src/server/routes/auth/callback/index.js

```diff
@@ -0,0 +1,32 @@
+import { createUserSession } from '#server/common/helpers/auth/create-user-session.js'
+import { randomUUID } from 'node:crypto'
+
+export default {
+  method: 'GET',
+  path: '/auth/callback',
+  options: {
+    auth: { strategy: 'entra-id', mode: 'try' }
+  },
+  handler: async function (request, h) {
+    if (!request.auth.isAuthenticated) {
+      return h.view('unauthorised')
+    }
+
+    const { profile, token } = request.auth.credentials
+
+    const { displayName = '' } = profile
+
+    const sessionId = randomUUID()
+
+    const userSession = {
+      sessionId,
+      displayName,
+      isAuthenticated: request.auth.isAuthenticated,
```

Comment:
Does the session data need `isAuthenticated` in it?

It looks like by line 14 `request.auth.isAuthenticated` will always be `true` (because of the early exit at line 12), thus `getUserSession` can only return an object if the user is authenticated